api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2055,12 +2056,43 @@ message UpstreamInventoryHello {
   // active.
   repeated string Services = 3 [(gogoproto.casttype) = "github.com/gravitational/teleport/api/types.SystemRole"];
   // Hostname is the hostname associated with the instance. This value is not required or guaranteed
-  // to be unique and its validity is not enforceable (i.e. join tokens do not constrian what an
+  // to be unique and its validity is not enforceable (i.e. join tokens do not constrain what an
   // instance can claim its hostname to be). This value exists only to assist users in correlating
   // instance resources with hosts.
   string Hostname = 4;
 }
 
+// UpstreamInventoryAgentMetadata is the message sent up the inventory control stream containing
+// metadata about the instance.
+message UpstreamInventoryAgentMetadata {
+  // Version advertises the teleport version of the instance.
+  string Version = 1;
+  // HostID advertises the ID of the instance.
+  string HostID = 2;
+  // Services advertises the currently live services of the instance that represent
+  // access protocols ("node", "kube", "app", "db" and "windows_desktop").
+  repeated string Services = 3;
```

Comment:
Nit: The `Hello` message already includes this data, and the hello is cached by the auth server for the lifetime of the stream.  Rather than double-sending the same data, I think it would be preferable to only send new data, and have the auth server construct the audit event from a combination of `Hello` and `AgentMetadata` params.

Ex:

```golang
func (c *Controller) handleAgentMetadata(handle *upstreamHandle, msg proto.UpstreamInventoryAgentMetadata) error {
    if err := c.auth.AnonymizeAndSubmit(&services.AgentMetadataEvent{
    	Version:               handle.Hello().Version,
 	HostId:                handle.Hello().ServerID,
 	Services:              handle.Hello().Services,
 	Os:                    msg.OS,
 	OsVersionInfo:         msg.OSVersionInfo,
 	HostArchitectureInfo:  msg.HostArchitectureInfo,
 	...
     }); err != nil {
 	log.Debugf("Unable to submit agent metadata: %v", err)
     }
}
```

Minor difference I know, but per-agent network load is a very serious scalability concern in large clusters.  Anything we can do to minimize load is a win.

Reply:
Thanks for the suggestion! Done in fa89947abf1917e6051d6107ee3ae94bc8bc9174.